### ISSUE TITLE
Improve violations doc to include the entire report format 

### ISSUE THREAD

**Issue by pranavgaikwad** [konveyor/analyzer-lsp#323]

**Comment by agoncal** [comment]
Thanks for the documentation on Violations: https://github.com/konveyor/analyzer-lsp/blob/main/docs/violations.md

Could `violations` have an identifier (just an incremental number) ? So we could have:

```yaml
  violations:
    id: 1
    chain-pom-001:
      description: ""
    id: 2
    chain-pom-002:
      description: ""
```

**Comment by shawn-hurley** [comment]
@agoncal I wonder if the ruleset and the rule name combination would work for the ID because they must be unique.

If that doesn't work, can you help us understand what you would use this ID for?
